crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -0,0 +1,248 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{
+    Arguments, CmpOp, Expr, ExprAttribute, ExprCall, ExprCompare, ExprContext, Identifier,
+};
+use ruff_python_semantic::{Modules, SemanticModel};
+use ruff_text_size::TextRange;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+///
+/// Checks for uses of the `re` module that can be replaced with builtin `str` methods.
+///
+/// ## Why is this bad?
+///
+/// Performing checks on strings directly can make the code simpler, may require
+/// less escaping, and will often be faster.
+///
+/// ## Example
+///
+/// ```python
+/// re.sub("abc", "", s)
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// s.replace("abc", "")
+/// ```
+///
+/// ## Details
+///
+/// Reports the following `re` calls when their first arguments are plain string
+/// literals, and no additional flags are passed:
+///
+/// - `sub`
+/// - `match`
+/// - `search`
+/// - `fullmatch`
+/// - `split`
+///
+/// For `sub`, the `repl` (replacement) argument must also be a string literal,
+/// not a function. For `match`, `search`, and `fullmatch`, the return value
+/// must also be used only for its truth value.
+///
+/// ## Fix safety
+///
+/// This rule's fix is marked as unsafe if the affected expression contains comments. Otherwise,
+/// the fix can be applied safely.
+#[derive(ViolationMetadata)]
+pub(crate) struct UnnecessaryRegularExpression {
+    replacement: String,
+}
+
+impl AlwaysFixableViolation for UnnecessaryRegularExpression {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Plain string pattern passed to `re` function".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        format!("Replace with `{}`", self.replacement)
+    }
+}
+
+/// RUF055
+pub(crate) fn unnecessary_regular_expression(checker: &mut Checker, call: &ExprCall) {
+    // adapted from unraw_re_pattern
+    let semantic = checker.semantic();
+
+    if !semantic.seen_module(Modules::RE) {
+        return;
+    }
+
+    let Some(qualified_name) = semantic.resolve_qualified_name(&call.func) else {
+        return;
+    };
+
+    let ["re", func] = qualified_name.segments() else {
+        return;
+    };
+
+    // skip calls with more than `pattern` and `string` arguments (and `repl`
+    // for `sub`)
+    let Some(re_func) = ReFunc::from_call_expr(semantic, call, func) else {
+        return;
+    };
+
+    // For now, restrict this rule to string literals
+    let Some(string_lit) = re_func.pattern.as_string_literal_expr() else {
+        return;
+    };
+
+    // For now, reject any regex metacharacters. Compare to the complete list
+    // from https://docs.python.org/3/howto/regex.html#matching-characters
+    let has_metacharacters = string_lit.value.to_str().contains([
+        '.', '^', '$', '*', '+', '?', '{', '}', '[', ']', '\\', '|', '(', ')',
```

Comment:
`]` and `}` can be removed from this list. They are only metacharacters when they follow `[` and `{`, which are already rejected here, so we can assume `]` and `}` are always matched literally.

Reply:
Fixed! I also removed `)`, which I think should behave the same way.